web/src/main/angular/src/app/core/components/apdex-score/apdex-score.component.ts

```diff
@@ -0,0 +1,80 @@
+import { Component, ComponentFactoryResolver, Injector, Input, OnInit, SimpleChanges } from '@angular/core';
+import { AnalyticsService, DynamicPopupService, TRACKED_EVENT_LIST } from 'app/shared/services';
+
+import { HELP_VIEWER_LIST, HelpViewerPopupContainerComponent } from 'app/core/components/help-viewer-popup/help-viewer-popup-container.component';
+
+const enum APDEX_SCORE_RANK {
+    EXCELLENT = 'excellent',
+    GOOD = 'good',
+    FAIR = 'fair',
+    POOR = 'poor',
+    UNACCEPTABLE = 'unacceptable',
+}
+
+@Component({
+    selector: 'pp-apdex-score',
+    templateUrl: './apdex-score.component.html',
+    styleUrls: ['./apdex-score.component.css']
+})
+export class ApdexScoreComponent implements OnInit {
```

Comment:
Add `OnChanges` lifecycle
```suggestion
export class ApdexScoreComponent implements OnInit, OnChanges {
```